unit_tests/log/test_main_log_page_fetcher.cpp

```diff
@@ -138,18 +138,17 @@ void test_env::on_receive_log_page (LOG_PAGEID page_id, const LOG_PAGE *log_page
   else
     {
       REQUIRE (error_code != NO_ERROR);
-      REQUIRE (log_page == nullptr);
     }
   g_log_page_fetcher_test_data.page_ids_requested[page_id].is_page_received = true;
-  delete log_page;
 }
 
-int log_reader::set_lsa_and_fetch_page (const log_lsa &lsa, fetch_mode fetch_page_mode)
+int
+logpb_fetch_page (THREAD_ENTRY *, const LOG_LSA *log_lsa, LOG_CS_ACCESS_MODE, LOG_PAGE *log_pgptr)
 {
-  std::unique_lock<std::mutex> lock (g_log_page_fetcher_test_data.map_mutex);
-  if (g_log_page_fetcher_test_data.page_ids_requested[lsa.pageid].require_log_page_valid)
+  if (g_log_page_fetcher_test_data.page_ids_requested[log_lsa->pageid].require_log_page_valid)
```

Comment:
shouldn't there be a
```
std::lock_guard<std::mutex> lock (g_log_page_fetcher_test_data.map_mutex);
```
here - as it was in the `log_reader::get_page` mock implementation?

Reply:
done